drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -170,6 +170,16 @@ int netdev2_ieee802154_set(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
     int res = -ENOTSUP;
 
     switch (opt) {
+        case NETOPT_CHANNEL:
+            /* validity needs to be checked by device, since sub-GHz and 2.4 GHz
+             * band radios have different legal values */
+            if (len > sizeof(dev->chan)) {
```

Comment:
The rest of the file doesn't use assert either yet. Let's do this in a seperate follow-up.